open-hackathon-server/src/hackathon/storage/azure_storage.py

```diff
@@ -22,13 +22,128 @@
 OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 """
+__author__ = 'ZGQ'
+
 from storage import Storage
+import json
+from uuid import uuid1
+from time import strftime
+from flask import g
+
+
+from hackathon import RequiredFeature
+from hackathon.constants import FILE_TYPE, HEALTH_STATUS, HEALTH
+
 
 __all__ = ["AzureStorage"]
 
 
 class AzureStorage(Storage):
-    """Hackathon file storage that saves all templates on MS azure"""
+    """Hackathon file storage that saves all templates on Azure storage
+
+    template files will be save at "<hostname>/<container_name>/<blob_name>"
+    uploaded images will be save at "<hostname>/<container_name>/<blob_name>"
+
+    """
+    cache_manager = RequiredFeature("cache_manager")
+    file_service = RequiredFeature("file_service")
+
+    def save(self, context):
+        """Save a file to Azure storage
+
+        :type context: Context
+        :param context: must have {"content":"***", "file_type":"***"} keys
+
+        :rtype context
```

Comment:
should be `:rtype: Context`.  `:` missed. and type should be `Context` rather than `context`